### PR TITLE
[presubmit] Enforce language attribute in project.yaml to be always set.

### PR DESCRIPTION
**Pull request by Dor1s**
My original intent was to enforce this for non C/C++ projects, but I couldn't come up with a good way to do it only in such cases. Landing this change might somewhat annoy people who contribute more changes to OSS-Fuzz for already integrated C/C++ projects, but that's fine with me -- yet another reason to remind people of the ideal integration.

**UPD:** when landing this change, I'll update all existing project.yaml files and we'll enforce the attribute from now on.

### CODE REVIEW

File: infra/presubmit.py [hunk 1]
```diff
@@ -190,12 +190,10 @@ def check_valid_emails(self):
   def check_valid_language(self):
     """Check that the language specified is valid."""
     language = self.data.get('language')
-    if not language:
-      return
-
     if language not in self.LANGUAGES_SUPPORTED:
```
Comment (inferno-chromium) [inline infra/presubmit.py]: Maybe add a better message when language is None
like "Missing language attribute" ?
Also, does our documentation for C/C++ mention the language attribute in template ?

Comment (Dor1s) [reply infra/presubmit.py]: > Also, does our documentation for C/C++ mention the language attribute in template ?

Nope, and I also need to test this before landing, if we agree that's an acceptable solution.

**Comment by jonathanmetzman** [comment]
> I wouldn't say im opposed to this, but it seems like a blunt solution for the issue of coverage builds happening for non-C++ projects (I don't know enough about the coverage infra to think of an alternative).
Offline max clarified that there are other reasons to do this, srcmap support is different for example, so this looks like a better solution than I initially thought.

**Comment by Dor1s** [comment]
Yeah, there are more use cases mentioned in https://github.com/google/oss-fuzz/issues/3297:
- different coverage logic
- different srcmap implementation
- default build configs for different languages
- `$LIB_FUZZING_ENGINE_DEPRECATED` vs `$LIB_FUZZING_ENGINE`, e.g. for Rust we can provide the deprecated version as a non-deprecated one since that's the intended way to use it
- (more speculative) bad build checks can differ for other languages

**Comment by Dor1s** [comment]
> When i tested, it appears that language: c++ is interpreted by pyyaml to be {'language': 'c++'}, so we don't need cpp.

I'll change `cpp` to `c++` and also update all existing projects so that the new presubmit check doesn't annoy users who will be making unrelated changes in OSS-Fuzz repo after we land this.

File: infra/presubmit.py (hunk 1 above)
Comment (Dor1s) [reply infra/presubmit.py]: done!

**Comment by Dor1s** [comment]
@oliverchang PTAL too, maybe you have another idea or any concerns regarding this one.

File: infra/templates.py [hunk 2]
```diff
@@ -13,9 +13,11 @@
 # limitations under the License.
 #
 ################################################################################
+"""Templates for OSS-Fuzz project files."""
 
 PROJECT_YAML_TEMPLATE = """\
 homepage: "<your_project_homepage>"
+language: <programming_language>
```
Comment (oliverchang) [inline infra/templates.py]: nit: give valid values as examples, as otherwise it's not clear what's expected (e.g. cpp or c++) ?

**Comment by Dor1s** [comment]
I've added `labels` and `selective_unpack` sections to presubmit script plus fixed a few files manually. There's still a dozen of presubmit errors (for missing primary_contact, non fuzzing engine (e.g. chakra), and some other custom value used only once), which I can't fix, so will land with them.

**Pull request merged**